src/com/sap/piper/GenerateStageDocumentation.groovy

```diff
@@ -0,0 +1,12 @@
+package com.sap.piper
+
+import java.lang.annotation.ElementType
+import java.lang.annotation.Retention
+import java.lang.annotation.RetentionPolicy
+import java.lang.annotation.Target
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target([ElementType.METHOD, ElementType.TYPE])
+public @interface GenerateStageDocumentation {
+    public String defaultStageName()
```

Comment:
What is the story behind having a stage name? Why is it not sufficient to rely on the name of the corresponding groovy file. For all the steps up to now we have that convention and at least for the steps this works fine.

Reply:
the name of the groovy file is the implementation, the defaultStageName is the default name of the stage in the pipeline. There is a significant difference here.